modules/samples/connectedcup/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.connectedcup.analytics-view/public/js/connectedcup.js

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var palette = new Rickshaw.Color.Palette({scheme: "classic9"});
+var graphMap = {};
+
+function drawGraph_connectedcup(from, to) {
+    var devices = $("#connectedcup-details").data("devices");
+    var tzOffset = new Date().getTimezoneOffset() * 60;
+
+    var streamIndex = 0;
+    var streams = ["temperature", "coffeelevel"];
+
+    populateGraph();
+
+    function populateGraph() {
+        if (streamIndex < 2) {
+            retrieveDataAndDrawLineGraph(streams[streamIndex], from, to);
+        }
+        streamIndex++;
+    }
+
+    function clearContent(type) {
+        $("#y_axis-" + type).html("");
+        $("#smoother-" + type).html("");
+        $("#legend-" + type).html("");
+        $("#chart-" + type).html("");
+        $("#x_axis-" + type).html("");
+        $("#slider-" + type).html("");
+    }
+
+    function initGraph(type, isMultilined) {
+        if (graphMap[type]) {
+            return graphMap[type];
+        }
+
+        var chartWrapperElmId = "#connectedcup-div-chart";
+        var graphWidth = $(chartWrapperElmId).width() - 50;
+
+        var graphConfig = {
+            element: document.getElementById("chart-" + type),
+            width: graphWidth,
+            height: 400,
+            strokeWidth: 2,
+            renderer: 'line',
+            interpolation: "linear",
+            unstack: true,
+            stack: false,
+            xScale: d3.time.scale(),
+            padding: {top: 0.2, left: 0.02, right: 0.02, bottom: 0.2},
+            series: []
+        };
+
+        if (devices) {
+            for (var i = 0; i < devices.length; i++) {
+                graphConfig['series'].push(
+                        {
+                            'color': palette.color(),
+                            'data': [{
```

Comment:
Single quote is the standard.
https://google.github.io/styleguide/javascriptguide.xml?showone=Strings#Strings